community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/exceptionHandlerFor2_2.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility
+
+import org.neo4j.cypher.internal.compiler.v2_2.spi.MapToPublicExceptions
+import org.neo4j.cypher.internal.compiler.v2_2.{CypherException => InternalCypherException}
+import org.neo4j.cypher.{ArithmeticException, CypherException => ExternalCypherException, CypherTypeException, EntityNotFoundException, FailedIndexException, HintException, IncomparableValuesException, IndexHintException, InternalException, InvalidArgumentException, InvalidSemanticsException, LabelScanHintException, LoadCsvStatusWrapCypherException, LoadExternalResourceException, MergeConstraintConflictException, NodeStillHasRelationshipsException, ParameterNotFoundException, ParameterWrongTypeException, PatternException, PeriodicCommitInOpenTransactionException, ProfilerStatisticsNotReadyException, SyntaxException, UniquePathNotUniqueException, UnknownLabelException, _}
+import org.neo4j.graphdb.Transaction
+
+object exceptionHandlerFor2_2 extends MapToPublicExceptions[ExternalCypherException] {
```

Comment:
This guy was extracted from `CompatibilityFor2_2` trait. It's `exceptionHandlerFor2_2 #handle(Throwable, Boolean, Transaction)` method is the main thing in this PR.
Also previously some methods of this object returned new exceptions and some threw those new exception. It feels that responsibility of this object is to *map* and not throw, so I've changed all methods to only return new exceptions.